### PR TITLE
Add CUDA graph NaN padding tests for attention, MoE, and per-token ops

### PR DESCRIPTION
**Pull request by tlrmchlsmth**
## Summary

CUDA graph NaN padding tests for the nvidia/DeepSeek-R1-0528-NVFP4-v2 production deployment (DP+EP with DeepEP LL, NVFP4 weights, FlashInfer CuteDSL experts).

### 1. FlashInfer NVFP4 quant kernel cross-row scale corruption (PRODUCTION IMPACT)

**`flashinfer.silu_and_mul_scaled_nvfp4_experts_quantize`** and **`flashinfer.scaled_fp4_grouped_quantize`** corrupt real token scales when padding rows contain NaN. This is the production kernel used between gate_up_proj and down_proj in the `FlashInferCuteDSLBatchedExperts` path.

Reproduction: few real tokens with CUDA graph padding (e.g., 8 real tokens in 64 slots) where padding rows have NaN from stale memory. The kernel scale computation leaks across rows, corrupting real token quantization scales.

**Impact**: Silent accuracy degradation, not NaN. The corruption produces wrong finite values (zeroed or incorrect scales) that compound across layers. Direct NaN in final output was not observed -- the GEMM chain produces wrong-but-finite results. However, this accumulated error across 61 layers could contribute to eventual NaN through numerical instability.

**Fix included**: Zero-fill padding rows in `flashinfer_cutedsl_batched_moe.py` before calling the FlashInfer quant kernels. Proven effective by `test_flashinfer_cutedsl_fix_prevents_corruption` which runs the wrapper with and without the fix.

Tests:
- `test_flashinfer_silu_quant_cross_row_corruption` -- direct kernel test (FAILS, proving bug exists)
- `test_flashinfer_grouped_quant_cross_row_corruption` -- direct kernel test (FAILS)
- `test_flashinfer_cutedsl_moe_masked_nan_padding` -- wrapper test (PASSES with fix)
- `test_flashinfer_cutedsl_fix_prevents_corruption` -- proves fix changes output
- `test_flashinfer_cutedsl_varying_masked_m` -- CUDA graph replay with shrinking batch size

### 2. vLLM silu_mul_cvt_fp16_to_fp4 cross-row scale corruption

**`silu_and_mul_scaled_fp4_experts_quant`** (vLLM CUDA kernel in `nvfp4_experts_quant.cu`) has the same cross-row corruption bug. Affects `CutlassExpertsFp4` path only (not production).

Tests: `test_fp4_quant_kernel_cross_row_scale_corruption` (xfail)

### 3. vLLM silu_mul_cvt_fp16_to_fp4 orphan row padding corruption

Padding rows beyond `expert_offsets[-1]` default to `expert_idx=0` and overwrite expert 0 scale. Latent -- no production code path creates the gap.

Tests: `test_fp4_quant_kernel_padding_corruption` (xfail)

### 4. Shared expert in-place mutation cross-stream race

`aux.wait_stream` does not prevent races from operations enqueued after the wait point. With `apply_router_weight_on_input=True` (Llama 4), `hidden.mul_()` on default stream races with shared expert reading `hidden` on aux stream.

Tests: `test_shared_experts_inplace_mutation_race` (xfail)

## What the test suite proves is safe

- All local per-token ops (RMSNorm, Linear, SiLU, RoPE, Embedding) correctly isolate NaN padding
- CUDA graph capture/replay with aggressive buffer pollution does not leak NaN through attention
- TRTLLM MLA decode kernel handles seq_len=0 padding requests cleanly
- KV cache null block (block 0) is not corrupted by padding tokens
- Production MLA layer with `num_actual_tokens=num_padded` (matching gpu_model_runner.py:2188) is safe
- FP8 KV cache write/read path handles NaN padding correctly
- Stale block table entries pointing to NaN-filled cache blocks do not corrupt real token attention output
- FlashInfer CuteDSL wrapper with the zero-fill fix handles NaN padding correctly

## Test coverage

**Single-GPU:**
- Standard attention backends (FlashAttn, FlashInfer, Triton) with CUDA graph + double-replay buffer pollution
- MLA decode backends (CUTLASS_MLA, FlashInfer_MLA, Triton_MLA) x auto/fp8 KV cache
- Full DeepSeek-R1 MLA layer with CUDA graph (bf16 + NVFP4 weights)
- Production-matching MLA test with `num_actual_tokens=num_padded`, FP8 KV cache, stale block tables
- Dense MLP with NVFP4 weights (ModelOptNvFp4Config)
- NVFP4 MoE with router (CutlassExpertsFp4) + CUDA graph
- Direct FlashInfer kernel corruption tests (silu_quant + grouped_quant)
- FlashInfer CuteDSL wrapper tests (with fix verification)
- Shared expert overlap with aux_stream (concurrent execution + race test)
- SM pressure tests (1000 CUDA graph replays with background GEMM contention)
- Numerical edge cases (zeros, moderate, tiny, identical inputs)
- NVFP4 weight edge cases (zero/huge/tiny/NaN scales, subnormal FP8, max FP4)
- Per-token ops (RMSNorm, Linear, SiLU, RoPE, Embedding)

**Multi-GPU DeepEP LL:**
- FP8 and NVFP4 dispatch at EP=2, EP=4, EP=8
- Asymmetric DP padding (one rank idle with pure NaN padding)
- EP=2 + DP=2 combination
- CUDA graph capture/replay + SM pressure + 1000 iterations

## Key testing techniques

- **Allocator poisoning**: `torch.cuda.empty_cache()` then fills allocator free list with 0xFF (NaN for fp8/fp16/bf16/fp32) at multiple block sizes
- **CUDA graph double-replay**: first replay fills ALL inputs with NaN to pollute every internal graph buffer; second replay restores real data and checks for corruption
- **SM pressure**: background GEMM on separate CUDA stream occupies ~20 SMs during 1000 replay iterations
- **Production-matching metadata**: `num_actual_tokens=num_padded`, `seq_lens=0` for padding, `slot_mapping=-1`, stale block table entries with NaN cache blocks
- **Fix verification**: runs wrapper with and without fix, asserts different output proves fix is active

## Running

```
.venv/bin/python -m pytest tests/v1/attention/test_cudagraph_nan_padding.py -v
```

### CODE REVIEW

File: tests/v1/attention/test_cudagraph_nan_padding.py [hunk 1]
```diff
@@ -0,0 +1,4185 @@
...
+        (32, 128, 64),
+        (8, 64, 8),      # few real, lots of padding
+        (4, 16, 1),       # extreme: 1 real token
+    ],
+    ids=["E4_m8", "E8_m16", "E8_m32", "E16_m64", "E32_m128",
+         "E8_m64_sparse", "E4_m16_1tok"],
+)
+def test_flashinfer_silu_quant_cross_row_corruption(
+    num_experts, m, num_real,
+):
```
Comment (tlrmchlsmth) [inline tests/v1/attention/test_cudagraph_nan_padding.py:1847]: @elvircrn this might be causing our NaNs

File: tests/v1/attention/test_cudagraph_nan_padding.py [hunk 2]
```diff
@@ -0,0 +1,4185 @@
...
+    """Test FlashInfer's scaled_fp4_grouped_quantize for cross-row
+    scale corruption with NaN in padding rows.
+
+    This is the production kernel for the FIRST quantization step
+    (hidden_states → NVFP4) in the FlashInferCuteDSLBatchedExperts
+    path when NVFP4 dispatch is NOT used. Uses CUDA graph
+    capture/replay with NaN pollution.
+    """
```
Comment (tlrmchlsmth) [inline tests/v1/attention/test_cudagraph_nan_padding.py:1958]: This also fails - but seems like it only affects us when we do NOT set `VLLM_DEEPEPLL_NVFP4_DISPATCH=1`

File: tests/v1/attention/test_cudagraph_nan_padding.py (hunk 1 above)
Comment (tlrmchlsmth) [reply tests/v1/attention/test_cudagraph_nan_padding.py:1847]:
```
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E4_m8] PASSED                                                                                                                                          [ 14%]
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E8_m16] PASSED                                                                                                                                         [ 28%]
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E8_m32] PASSED                                                                                                                                         [ 42%]
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E16_m64] PASSED                                                                                                                                        [ 57%]
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E32_m128] PASSED                                                                                                                                       [ 71%]
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E8_m64_sparse] FAILED                                                                                                                                  [ 85%]
tests/v1/attention/test_cudagraph_nan_padding.py::test_flashinfer_silu_quant_cross_row_corruption[E4_m16_1tok] PASSED                                                                                                                                    [100%]
```
